async/src/connection.rs

```diff
@@ -541,14 +541,22 @@ impl Connection {
       channel.state.clone()
     }).unwrap();
     if let ChannelState::WillReceiveContent(queue_name, consumer_tag) = state {
-      self.set_channel_state(channel_id, ChannelState::ReceivingContent(queue_name.clone(), consumer_tag.clone(), size as usize));
+      if size > 0 {
+        self.set_channel_state(channel_id, ChannelState::ReceivingContent(queue_name.clone(), consumer_tag.clone(), size as usize));
+      } else {
+        self.set_channel_state(channel_id, ChannelState::Connected);
+      }
       if let Some(ref mut c) = self.channels.get_mut(&channel_id) {
         if let Some(ref mut q) = c.queues.get_mut(&queue_name) {
           if let Some(ref consumer_tag) = consumer_tag {
             if let Some(ref mut cs) = q.consumers.get_mut(consumer_tag) {
               if let Some(msg) = cs.current_message.as_mut() {
                 msg.properties = properties;
               }
+              if size == 0 {
+                let message = cs.current_message.take().expect("there should be an in flight message in the consumer");
+                cs.messages.push_back(message);
+              }
```

Comment:
I think you're missing the same block for basic get just a couple of lines below?

Reply:
Right, should be good now.